### DISCUSSION TITLE
Custom stdin/stdout ?

### DISCUSSION THREAD

**Discussion by KaKi87** [SBoudrias/Inquirer.js#1183] [Q&A] [answered]
Hello,

Is it possible to use `inquirer` with other `Stream` instances than `process.stdin` & `process.stdout` ?

Thanks

**Comment by SBoudrias** [comment, marked as answer]
Hey @KaKi87, this is much easier to do with the new (beta) prompts. All those packages are published under the `@inquirer/*` namespace.

**Reply by KaKi87** [reply]
Thanks for answering, how do I get started ?

**Reply by KaKi87** [reply]
Hello ?

**Reply by SBoudrias** [reply]
You'll want to use the second argument:

```js
import input from '@inquirer/input';

const answer = await input(
    { message: 'Package name' },
    { input, output }
);
```

You can see all packages here https://github.com/SBoudrias/Inquirer.js/tree/master/packages and some beginning documentation in most folders.

**Reply by KaKi87** [reply]
Thanks !
One more thing : can I make it write `\r\n` instead of `\n` when pressing Enter ?

**Reply by SBoudrias** [reply]
@KaKi87 this is unfortunately built-in Node. As a workaround, I'd recommend calling `answer.replace(...)` on the answer.

**Reply by KaKi87** [reply]
Actually, I have a more important issue : right after using any prompt, my streams `finish` events are fired :thinking:

**Reply by SBoudrias** [reply]
This likely comes from this line here: https://github.com/SBoudrias/Inquirer.js/blob/master/packages/core/src/lib/screen-manager.ts#L105

I'm happy to revise; this is "new" code in git, but it's coming from the first few implementations ~10y ago where the Node.js built-in readline was far from stable.

I'll just want us to test on a few OS & Node version to make sure it works as expected without.

**Reply by KaKi87** [reply]
How is it going ?
